packages/cli-repl/src/startup-timing.ts

```diff
@@ -1,30 +1,54 @@
-const jsTimingEntries: [string, bigint][] = [];
-const timing: {
-  markTime: (label: string) => void;
-  getTimingData: () => [string, number][];
-} = !process.env.MONGOSH_SHOW_TIMING_DATA
-  ? {
-      markTime: () => {
-        /* ignore */
-      },
-      getTimingData: () => [],
-    }
-  : (process as any).boxednode
-  ? {
-      markTime: (process as any).boxednode.markTime,
-      getTimingData: (process as any).boxednode.getTimingData,
-    }
-  : {
-      markTime: (label: string) => {
-        jsTimingEntries.push([label, process.hrtime.bigint()]);
-      },
-      getTimingData: () => {
-        const data = jsTimingEntries.sort((a, b) => Number(a[1] - b[1]));
-        // Adjust times so that process initialization happens at time 0
-        return data.map(([label, time]) => [label, Number(time - data[0][1])]);
-      },
+import { TimingCategories, type TimingInterface } from '@mongosh/types';
+
+const jsTimingEntries: [string, string, bigint][] = [];
+
+function linkTimingInterface(): TimingInterface {
+  const boxedNode = (process as any).boxednode;
+
+  // If we are bundled with boxednode, just use the provided interface
+  if (boxedNode) {
+    return {
+      markTime: boxedNode.markTime,
+      getTimingData: boxedNode.getTimingData,
     };
+  }
+
+  // Otherwise, use a JS implementation (mostly for development)
+  return {
+    markTime: (category, label) =>
+      jsTimingEntries.push([category, label, process.hrtime.bigint()]),
+    getTimingData: () => {
+      const data = jsTimingEntries.sort((a, b) => Number(a[2] - b[2]));
+      // Adjust times so that process initialization happens at time 0
+      return data.map(([category, label, time]) => [
+        category,
+        label,
+        Number(time - data[0][2]),
+      ]);
+    },
+  };
+}
+
+export function summariseTimingData(timingData: [string, string, number][]): {
```

Comment:
the test in `startup-timing.spec.ts` implies that `timingData` is always of type `[TimingCategory, string, number][]`.  Is that correct?  If so - can we make the type stricter here?

Reply:
We could, but it would be misleading. The thing is that when bundle the application with BoxedNode, timingData can have also strings not included in TimingCategory (it's a C++ library we are linking). Summarise Timing Data should work for both the cases it's using the BoxedNode implementation of getTimingData and the TypeScript version (in the dev environment).

Reply:
Just to clarify: the only reason TimingCategory exists is to make sure that at least, on the typescript side of things (99% of times) we have a clear convention on what timing categories do. But in BoxedNode we might need sometimes a few more categories, as it does all the bootstrapping.

Reply:
That makes sense but it's sort of misleading that `linkTimingInterface` uses a `TimingInterface` and that when boxednode is linked, it supposedly satisfies that interface.  

Not that important, but this is the kind of "Typescript lie" that I like to avoid because it causes confusion for future devs, and can lead to scenarios where bugs are introduced because developers rely on the types

Reply:
We can use TimingCategory | string if you think it's better in summariseTimingData. Semantics are the same, but it's more clear for a developer that this can be either the enum or other unknown strings from other sources.

WDYT?